ts-tests/integration-tests/precompile-contract.test.ts

```diff
@@ -229,28 +265,71 @@ describeLitentry('Test Parachain Precompile Contract', ``, (context) => {
+            'executeDelegationRequest'
+        );
         const { data: balanceAfterRevoke } = await context.api.query.system.account(evmAccountRaw.mappedAddress);
         expect(balanceAfterRevoke.free.toNumber()).to.closeTo(balance.free.toNumber(), toBigNumber(1));
         expect(balanceAfterRevoke.reserved.toNumber()).to.eq(0);
 
         // delegate(collator, amount);
-        const delegate = precompileContract.methods.delegate(collatorPublicKey, toBigNumber(57));
-        await executeTransaction(delegate, 'delegate');
+        const delegate = precompileStakingContract.methods.delegate(collatorPublicKey, toBigNumber(57));
+        await executeTransaction(delegate, precompileStakingContractAddress, 'delegate');
         const { data: balanceAfterDelegate } = await context.api.query.system.account(evmAccountRaw.mappedAddress);
         expect(balanceAfterDelegate.reserved.toNumber()).to.eq(toBigNumber(57));
 
         // In case evm is not enabled in Normal Mode, switch back to filterMode, after test.
         let extrinsic = context.api.tx.sudo.sudo(context.api.tx.extrinsicFilter.setMode(filterMode));
         await signAndSend(extrinsic, context.alice);
 
-        console.timeEnd('Test precompile contract');
+        console.timeEnd('Test precompile staking contract');
+    });
+    step('Test precompile bridge contract', async function () {
+        let balance = (await context.api.query.system.account(evmAccountRaw.mappedAddress)).data;
+        if (balance.free.toNumber() < toBigNumber(0.01)) {
+            await transferTokens(context.alice, evmAccountRaw);
+            expect(balance.free.toNumber()).to.gt(toBigNumber(0.01));
+        }
+
+        // update chain bridge fee
+        const updateFeeTx = context.api.tx.sudo.sudo(context.api.tx.chainBridge.updateFee(0, bn1e12 / 1000));
+        await signAndSend(updateFeeTx, context.alice);
+
+        const bridge_fee = await context.api.query.chainBridge.bridgeFee(0);
+        expect(bridge_fee.toString()).to.eq((bn1e12 / 1000).toString());
+
+        // set chainId to whitelist
+        const whitelistChainTx = context.api.tx.sudo.sudo(context.api.tx.chainBridge.whitelistChain(0));
+        await signAndSend(whitelistChainTx, context.alice);
+
+        // The above two steps are necessary, otherwise the contract transaction will be reverted.
+        // transfer native token
+        const transferNativeTx = precompileBridgeContract.methods.transferNative(
+            bn1e12 / 100, // 0.01 LIT
+            '0xaaafB3972B05630fCceE866eC69CdADd9baC2772', // random address
+            0
+        );
+
+        const res = await executeTransaction(transferNativeTx, precompileBridgeContractAddress, 'transferNative');
+        expect(res.status).to.eq(true);
+
+        const eventsPromise = subscribeToEvents('chainBridge', 'FungibleTransfer', context.api);
+        const events = (await eventsPromise).map(({ event }) => event);
+        expect(events.length).to.eq(1);
```

Comment:
It might be better to compare the event content though

Reply:
Added checks for event data.